__tests__/app.test.js

```diff
@@ -105,29 +105,159 @@ beforeEach(() => {
+         
+          articles.forEach((comment) => {
+            expect(comment).toEqual(
+              expect.objectContaining({
+              
+                article_id: expect.any(Number),
+                comment_id: expect.any(Number),
+body:expect.any(String),
+                author: expect.any(String),
+                created_at: expect.any(String),
+                votes: expect.any(Number)
+              }))
+            })
+            
+            })
+          })
+
+          test('Response 400 and appropriate message if article_id is a number but article does not exist', () => {
+            return request(app)
+                .get('/api/articles/10000000')
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.message).toEqual('Bad Request')
+                })
+          })
+          
+          test('Response 400 and appropriate message if article_id is not a number', () => {
+            return request(app)
+                .get('/api/articles/banana')
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.message).toEqual('Bad Request')
+                })
+          })
+          
+        
   })
 
   
+
+  describe('POST /api/articles/:article_id/comments', () => {
+    it.only('Response 201 and returns newly inserted comment in an object with key of comment', () => {
+        const newComment = {
+            username: `butter_bridge`,
+            body: `Posting a new comment`
+        }
+        return request(app)
+            .post('/api/articles/2/comments')
+            .send(newComment)
+            .expect(201)
+            .then(({ body }) => {
+                expect(body.comment[0]).toEqual({
+                    article_id: 2,
+                    author: "butter_bridge",
+                    body: "Posting a new comment",
+                    comment_id: 19,
+                    created_at: expect.any(String),
+                    votes: 0,
+                })
+            })
+    })
+
+    it('should respond with a status 400 Bad request', ()=>{
+      return request(app)
+      .post('/api/articles/15/comments')
+      .send({username: 'butter_bridge', body: 'Posting a new comment'})
+      .expect(400)
+      .then(({ body }) => {
+          expect(body.msg).toBe('Bad request');
+      })
+    })
+  
+    it('Response 400 and appropriate message if article_id is a number but article does not exist', () => {
+      return request(app)
+          .get('/api/articles/10000000')
+          .expect(400)
+          .then(({ body }) => {
+              expect(body.msg).toEqual('Bad Request')
+          })
+    })
+
+    it('200: GET - responds with server ok message', () => {
+      return request(app)
+        .get('/api')
+        .expect(200)
+        .then((response) => {
+          console.log(response.body, 'RESPONSE BODY');
+          expect(response.body.msg).toBe('Server is OK');
+        });
+    });
+
```

Comment:
Missing 404 test to post a valid comment but with a username that doesn't exist in the database

for example {username: "mahmud", body: "valid comment"}